From 59b0770ca51fd556db4417bf7d409259cb725a83 Mon Sep 17 00:00:00 2001
From: Alex Williamson <alex.williamson@redhat.com>
Date: Mon, 17 Oct 2011 03:28:23 +0200
Subject: [PATCH 2/2] Fix DEBUG builds for filenames with hyphens

RH-Author: Alex Williamson <alex.williamson@redhat.com>
Message-id: <20111017032752.14510.47001.stgit@s20.home>
Patchwork-id: 34280
O-Subject: [RHEL6.2 gpxe PATCH 2/1] [build] Fix DEBUG builds for filenames with hyphens
Bugzilla: 743893
RH-Acked-by: Xiao Wang <jasowang@redhat.com>
RH-Acked-by: Amos Kong <akong@redhat.com>
RH-Acked-by: Michael S. Tsirkin <mst@redhat.com>

From: Joshua Oreman <oremanj@rwcr.net>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=743893
Upstream commit: gpxe.git 3fa277920804b056ce72fd9ba0074a7ff7c871fa
RHEL note: tested with and without DEBUG="virtio-net" build.  This is
for ease of future debugging only and does not change non-debug build.

Debug builds for filenames with hyphens such as:

     $ make bin/via-rhine.dsk DEBUG=via-rhine

fail with:

     [BUILD] bin/via-rhine.dbg1.o
     <command-line>: error: missing whitespace after the macro name
     make: *** [bin/via-rhine.dbg1.o] Error 1

This is because "-" is not a legal character in C identifiers, and
gcc rejects "-Ddebug_via-rhine=1" as an argument.

Signed-off-by: Daniel Verkamp <daniel@drv.nu>
Signed-off-by: Joshua Oreman <oremanj@rwcr.net>
Signed-off-by: Marty Connor <mdc@etherboot.org>
---

Per suggestion from Jason, add trivial debugging fix to series.  Debug
output, specified via DEBUG env during make, cannot include hyphenated
files without this fix.  The new virtio-net driver includes some debug
output support, so this will facilitate turning it on for future
debugging.

 src/Makefile.housekeeping |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Signed-off-by: Michal Novotny <mignov@gmail.com>
---
 src/Makefile.housekeeping |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/Makefile.housekeeping b/src/Makefile.housekeeping
index 2ab842e..6bbd13f 100644
--- a/src/Makefile.housekeeping
+++ b/src/Makefile.housekeeping
@@ -318,7 +318,7 @@ $(BIN)/%.flags :
 #
 COMPILE_c	= $(CC) $(CFLAGS) $(CFLAGS_c) $(OBJ_CFLAGS)
 RULE_c		= $(Q)$(COMPILE_c) -c $< -o $@
-RULE_c_to_dbg%.o = $(Q)$(COMPILE_c) -Ddebug_$(OBJECT)=$* -c $< -o $@
+RULE_c_to_dbg%.o = $(Q)$(COMPILE_c) -Ddebug_$(subst -,_,$(OBJECT))=$* -c $< -o $@
 RULE_c_to_c	= $(Q)$(COMPILE_c) -E -c $< > $@
 RULE_c_to_s	= $(Q)$(COMPILE_c) -S -g0 -c $< -o $@
 
-- 
1.7.4.4

